### DISCUSSION TITLE
Problem of warmup_ratio

### DISCUSSION THREAD

**Discussion by Icecream-blue-sky** [open-mmlab/mmdetection#9268] [Q&A]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmdetection/issues) and [Discussions](https://github.com/open-mmlab/mmdetection/discussions) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version (master)](https://github.com/open-mmlab/mmdetection) or [latest version (3.x)](https://github.com/open-mmlab/mmdetection/tree/dev-3.x).

### Task

I'm using the official example scripts/configs for the officially supported tasks/models/datasets.

### Branch

master branch https://github.com/open-mmlab/mmdetection

### Environment

python3

### Reproduces the problem - code sample

The warmup_ratio means "the ratio of the starting learning rate used for warmup". But I find that if I set the lr to 0.1 and warmup_ratio to 0.001, the actual starting learning rate used for warmup is not 0.001 * 0.1, why? How to caculate the actual starting learning rate?
![image](https://user-images.githubusercontent.com/26771037/200329044-472f5fe7-816d-4ee1-a922-8150f925fe62.png)

### Reproduces the problem - command or script

None

### Reproduces the problem - error message

None

### Additional information

None

**Comment by ZwwWayne** [comment]
This is because the log shows the LR at the 50th iteration rather than the first iteration.

**Comment by Icecream-blue-sky** [comment]
> This is because the log shows the LR at the 50th iteration rather than the first iteration.

Set x1=1, y1=0.001\*0.1 (the learning rate of the first iteration), x2=1000, y2=0.01 (the initial learning rate), so the function of linear warmming up is  y = 9.91\*10^(-6)\*x + 9.009\*10^(-5). So when x=50 (the 50th iteration), the learning rate y should be 5.8559\*10^(-4), which is different from the one in the log, why?

**Comment by ZwwWayne** [comment]
The calculation is implemented here: https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/hooks/lr_updater.py#L87
